tests/test_web_exceptions.py

```diff
@@ -172,3 +173,14 @@ def test_link_header_451(buf, request):
 
     assert 'http://warning.or.kr/' == resp.link
     assert '<http://warning.or.kr/>; rel="blocked-by"' == resp.headers['Link']
+
+
+def test_HTTPException_retains_cause():
+    with pytest.raises(web.HTTPException) as ei:
+        try:
+            raise Exception('CustomException')
+        except Exception as exc:
+            raise web.HTTPException() from exc
+    tb = ''.join(format_exception(ei.type, ei.value, ei.tb))
```

Comment:
is `ei` available outside of `with` block?

Reply:
Yes it is.
We use it widely in the test suite:
```
    with pytest.raises(aiohttp.ClientResponseError) as info:
        await response.start(connection)

    assert info.value.request_info is request_info
```